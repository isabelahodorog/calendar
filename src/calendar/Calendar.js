import React from 'react';
import PropTypes from 'prop-types';

import Day from './day';
import styles from './Calendar.module.css'

const Calendar = ({currentDate}) => {
    const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
    const currentMonth = months[currentDate.getMonth()];

    const firstDayOfMonth = new Date(currentDate.getFullYear(), currentDate.getMonth(), 1);
    const lastDayOfMonth = new Date(currentDate.getFullYear(), currentDate.getMonth() + 1, 0);

    let calendarDays = [];
    calendarDays.push(<Day day={1} isCurrent={1===currentDate.getDate()} key={1} columnStart={firstDayOfMonth.getDay()}/>)

    for (let i = 2; i <= lastDayOfMonth.getDate(); i++) {
        calendarDays.push(<Day day={i} isCurrent={i===currentDate.getDate()} key={i}/>)
    }

    return (
        <div className={styles['Calendar']}>
            <h1>
                {currentMonth} {currentDate.getFullYear()}
            </h1>
            <div className={styles['Calendar--grid']}>
                <span>Mo</span>
                <span>Tu</span>
                <span>We</span>
                <span>Th</span>
                <span>Fr</span>
                <span>Sa</span>
                <span>Su</span>
            </div>
            <hr/>
            <div className={styles['Calendar--grid']}>
                {calendarDays}
            </div>
        </div>
    );
};

Calendar.propTypes = {
    currentDate: PropTypes.instanceOf(Date).isRequired
}

export default Calendar;