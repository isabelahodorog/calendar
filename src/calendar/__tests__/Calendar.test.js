import React from 'react';

import { render } from '@testing-library/react';

import Calendar from '../Calendar';

describe('Calendar: ', () => {
    it('should match default snapshot', () => {
        const { container } = render(<Calendar currentDate={new Date(2020, 6, 17)}/>);
        expect(container).toMatchSnapshot();
    });

    it('should be displayed as a grid starting from third column', () => {
        const testDate = new Date(2020, 6, 6); //july 2020 starts on a Wednesday
        const { getAllByTestId } = render(<Calendar currentDate={testDate} />);
        expect(getAllByTestId("calendarDay")[0].className).toBe("Day Day--gridItem-colStart3");
    });
})