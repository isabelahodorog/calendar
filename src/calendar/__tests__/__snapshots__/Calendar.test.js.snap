// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar:  should match default snapshot 1`] = `
<div>
  <div
    class="Calendar"
  >
    <h1>
      July
       
      2020
    </h1>
    <div
      class="Calendar--grid"
    >
      <span>
        Mo
      </span>
      <span>
        Tu
      </span>
      <span>
        We
      </span>
      <span>
        Th
      </span>
      <span>
        Fr
      </span>
      <span>
        Sa
      </span>
      <span>
        Su
      </span>
    </div>
    <hr />
    <div
      class="Calendar--grid"
    >
      <div
        class="Day Day--gridItem-colStart3"
        data-testid="calendarDay"
      >
         
        1
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        2
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        3
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        4
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        5
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        6
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        7
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        8
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        9
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        10
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        11
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        12
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        13
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        14
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        15
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        16
         
      </div>
      <div
        class="Day Day--isCurrent"
        data-testid="calendarDay"
      >
         
        17
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        18
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        19
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        20
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        21
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        22
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        23
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        24
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        25
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        26
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        27
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        28
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        29
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        30
         
      </div>
      <div
        class="Day"
        data-testid="calendarDay"
      >
         
        31
         
      </div>
    </div>
  </div>
</div>
`;
